### PR TITLE
Fix the issue current class is not being added to the <li>

### LINKED ISSUE #1608: current class is not being added to the <li>
**Describe the bug**

When you set a default tab in Profile page Appearance > Customize > BuddyBoss Platform > Profile Navigation, whatever tab you set as default will have the last part of the URL stripped when you visit it.
https://prnt.sc/tmdg6s

Due to this, it doesn't add the current class to the li so it doesn't highlight the link in the buddypanel.
https://prnt.sc/tmddkb

**To Reproduce**

1. Go to https://demos.buddyboss.com/online-communities
2. Click on 'my timeline' menu item from buddypanel
3. you will be redirected to your timeline.
4. But the menu in the buddypanel doesn't highlight.

**Expected behavior**
It should add the current class in the buddypanel menu if visiting the default tab of profile navigation.

**Screenshots**
https://prnt.sc/tmdg6s
https://prnt.sc/tmddkb

**Support ticket links**
https://secure.helpscout.net/conversation/1230761378/84213/
https://secure.helpscout.net/conversation/1293201370/100707/

**Ref:** https://github.com/buddyboss/buddyboss-theme/issues/872

### PR DESCRIPTION
**Pull request by krunal-buddyboss**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #1608 

### How to test the changes in this Pull Request:

1. Go to https://demos.buddyboss.com/online-communities
2. Click on 'my timeline' menu item from buddypanel
3. you will be redirected to your timeline.
4. But the menu in the buddypanel doesn't highlight.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-core/bp-core-filters.php [hunk 1]
```diff
@@ -780,6 +780,29 @@ function bp_setup_nav_menu_item( $menu_item ) {
 			} else {
 				$menu_item->classes = array( 'current_page_item', 'current-menu-item' );
 			}
+		} else {
+			$domain_arr = array( bp_loggedin_user_domain() );
+			if ( in_array( $current, $domain_arr ) ) {
```
Comment (chetansatasiya) [inline src/bp-core/bp-core-filters.php:785]: @krunal-buddyboss No need seprate variable for this.

```suggestion
			if ( in_array( $current, array( bp_loggedin_user_domain() ) ) ) {
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]

File: src/bp-core/bp-core-filters.php [hunk 2]
```diff
@@ -780,6 +780,28 @@ function bp_setup_nav_menu_item( $menu_item ) {
...
+						$component = 'xprofile';
+					}
+					if ( bp_is_active( $component ) ) {
+						if ( strpos( $menu_item->url, $tab ) !== false ) {
+							if ( is_array( $menu_item->classes ) ) {
+								$menu_item->classes[] = 'current_page_item';
+								$menu_item->classes[] = 'current-menu-item';
+							} else {
+								$menu_item->classes = array( 'current_page_item', 'current-menu-item' );
+							}
```
Comment (dipskakadiya) [inline src/bp-core/bp-core-filters.php:800]:
```suggestion
                                                                  $menu_item->classes = is_array( $menu_item->classes ) ? $menu_item->classes : array() ;
								$menu_item->classes[] = 'current_page_item';
								$menu_item->classes[] = 'current-menu-item';
							
```

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

**Review by dipskakadiya — APPROVED** [review]
Good to merged 👍

**Review by dipskakadiya — APPROVED** [review]
Good to merged 👍

**Comment by ankitguptaindia** [comment]
@dipskakadiya @sourabhmatolia Working on this. 
Will Add final update in morning.

**Comment by ankitguptaindia** [comment]
@sourabhmatolia 

Above mentioned issue has been fixed. 
https://www.loom.com/share/fab54d48250b41f49fed2c40ec53956e

Ready to Merge 🚀

**Review by ankitguptaindia — APPROVED** [review]
**Closing Remark-** 

- [x] Read PR description

- [x] Checked Description on related Issue

- [x] Pulled respective branch and checked module by pull stable branch (Checked the Before Fix & After Fix differences)

- [ ] Tested negative test scenario

- [ ] Updated Labels as per current status

**Final Update-** Ready to merge 🚀

**Review by sourabhmatolia — APPROVED** [review]

**Pull request merged**